taca/analysis/analysis_nanopore.py

```diff
@@ -217,35 +218,66 @@ def process_minion_delivery_run(minion_run):
+            logger.debug(f"Run {ont_run.run_id} does not exist in the database, creating entry for ongoing run.")
             
             # Create an ongoing run document
             run_path_file = os.path.join(ont_run.run_dir, 'run_path.txt')
             sesh.create_ongoing_run(ont_run, open(run_path_file, "r").read().strip())
+            logger.debug(f"Successfully created db entry for ongoing run {ont_run.run_id}.")
 
-        # If run is finished and an ongoing run document exists
-        if len(ont_run.summary_file) != 0 and sesh.check_run_status(ont_run) == "ongoing":
+        # If a run document DOES exist in the database
+        else:
+            # If the run document is marked as "ongoing"
+            if sesh.check_run_status(ont_run) == "ongoing":
+                
+                logger.debug(f"Run {ont_run.run_id} exists in the database as an ongoing run.")
 
-            # Parse the MinKNOW .json and .html report files and finish the ongoing run document
-            glob_json = glob.glob(ont_run.run_dir + '/report*.json')
-            glob_html = glob.glob(ont_run.run_dir + '/report*.html')
-            
-            dict_json = json.load(open(glob_json[0], "r"))
-            dict_html = {
-                "minknow_report_name": glob_html[0].split("/")[-1],
-                "minknow_report_content": html.escape(open(glob_html[0], "r").read())
-            }
+                # If the run is finished
+                if len(ont_run.summary_file) != 0:
+
+                    logger.debug(f"Run {ont_run.run_id} has finished sequencing, updating the db entry.")
+
+                    # Parse the MinKNOW .json and .html report files and finish the ongoing run document
+                    glob_json = glob.glob(ont_run.run_dir + '/report*.json')
+                    glob_html = glob.glob(ont_run.run_dir + '/report*.html')
 
-            sesh.finish_ongoing_run(ont_run, dict_json, dict_html)
+                    if len(glob_json) == 0 or len(glob_html) == 0:
+                        logger.error(f"Run {ont_run.run_id} is marked as finished, but missing report files.")
+                        raise AssertionError
+                    
+                    dict_json = json.load(open(glob_json[0], "r"))
+                    dict_html = {
+                        "minknow_report_name": glob_html[0].split("/")[-1],
+                        "minknow_report_content": html.escape(open(glob_html[0], "r").read())
+                    }
 
+                    sesh.finish_ongoing_run(ont_run, dict_json, dict_html)
+                    logger.debug(f"Successfully updated the db entry of run {ont_run.run_id}")
+
+                else:
+                    logger.debug(f"Run {ont_run.run_id} has not finished sequencing, do nothing.")
+            
+            else:
+                logger.debug(f"Run {ont_run.run_id} exists in the database as an finished run, do nothing.")
+        
         return True
+
     except:
```

Comment:
Maybe catch a specific error here, like AssertionError in this case? Otherwise, if there are other errors they will be harder to troubleshoot

Reply:
If a predefined error is caught within this function, it would already have been added to the taca log.

Reply:
Ok, so it's intended as a catch-all exception? Maybe at least log this undefined error as well?

Reply:
Yeah, but if it's not a predefined error, we'd not get any indication of what the error is so troubleshooting would be harder

Reply:
Yes so something like 
```
except Exception as e:
    log.error(e)
    return False
```
Warning, untested code from top of my mind.

Reply:
Added more thorough error-tracing in [d2b6119](https://github.com/SciLifeLab/TACA/pull/350/commits/d2b6119c370fdf795788d50d6c224e9ac7c4a0ab)